egs/ljspeech/tts1/run.sh

```diff
@@ -228,3 +228,91 @@ if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ]; then
     [ ${i} -gt 0 ] && echo "$0: ${i} background jobs are failed." && false
     echo "Finished."
 fi
+
+feat_tr_dir=${dumpdir}/${train_set}_kd; mkdir -p ${feat_tr_dir}
+feat_dt_dir=${dumpdir}/${dev_set}_kd; mkdir -p ${feat_dt_dir}
+if [ ${n_average} -gt 0 ]; then
+    model=model.last${n_average}.avg.best
+fi
+outdir=${expdir}/outputs_${model}_$(basename ${decode_config%.*})
+if [ ${stage} -le 6 ] && [ ${stop_stage} -ge 6 ]; then
+    echo "stage 6: Decoding"
+    if [ ${n_average} -gt 0 ]; then
+        average_checkpoints.py --backend ${backend} \
+                               --snapshots ${expdir}/results/snapshot.ep.* \
+                               --out ${expdir}/results/${model} \
+                               --num ${n_average}
+    fi
+    pids=() # initialize pids
+    for name in ${dev_set} ${train_set}; do
+    (
+        [ ! -e ${outdir}/${name} ] && mkdir -p ${outdir}/${name}
+        cp ${dumpdir}/${name}/data.json ${outdir}/${name}
+        splitjson.py --parts ${nj} ${outdir}/${name}/data.json
+        # decode in parallel
+        ${train_cmd} --num-threads 2 JOB=1:${nj} ${outdir}/${name}/log/decode.JOB.log \
+            tts_decode.py \
+                --backend ${backend} \
+                --ngpu 0 \
+                --verbose ${verbose} \
+                --out ${outdir}/${name}/feats.JOB \
+                --json ${outdir}/${name}/split${nj}utt/data.JOB.json \
+                --model ${expdir}/results/${model} \
+                --save-durations true \
+                --config ${decode_config}
+        # concatenate scp files
+        for n in $(seq ${nj}); do
+            cat "${outdir}/${name}/feats.$n.scp" || exit 1;
+        done > ${outdir}/${name}/feats.scp
+        for n in $(seq ${nj}); do
+            cat "${outdir}/${name}/durations.$n.scp" || exit 1;
+        done > ${outdir}/${name}/durations.scp
```

Comment:
We may want to add logic to remove bad samples (e.g. contains deletions and repetitions). Possible solutions:
1) Check focus rate to detect the repeat or deletion (probably easier)
2) Use ASR-based objective evaluation (i.e. use utterances whose CER=0.0) (code will be dirty...)

Reply:
Currently, I removed the samples reaching the maximum length.